sim/battle.ts

```diff
@@ -2393,15 +2410,19 @@ export class Battle extends Dex.ModdedDex {
 			}
 		}
 
-		if (!this.p1.pokemonLeft && !this.p2.pokemonLeft) {
+		let team1PokemonLeft = this.p1.pokemonLeft;
+		if (this.p1.ally) team1PokemonLeft += this.p1.ally.pokemonLeft;
+		let team2PokemonLeft = this.p2.pokemonLeft;
+		if (this.p2.ally) team2PokemonLeft += this.p2.ally.pokemonLeft;
+		if (!team1PokemonLeft && !team2PokemonLeft) {
```

Comment:
This fails for free-for-all battles.

Reply:
Free-for-all battles, or as they're actually called, "Battle Royals", have their own unique win condition. https://bulbapedia.bulbagarden.net/wiki/Battle_Royal#Scoring

Reply:
(Not to be confused with [Battle Royales](https://bulbapedia.bulbagarden.net/wiki/Battle_Royale))

Reply:
Either way, this should be rewritten to use the `% 2` approach, instead of `.ally`.

Reply:
Ok, so I didn’t realize that “free-for-all” meant a multi battle where you compete with your “teammate” and didn’t mean Battle Royal.